R/checkers.R

```diff
@@ -35,7 +46,8 @@ check_elements_non_neg <- function(x, arg = "x", call = rlang::caller_env()) {
   is_non_neg <- (x >= 0) | is.na(x)
   if (!all(is_non_neg)) {
     cli::cli_abort(
-      c("{.arg {arg}} has negative elements",
+      c("{.arg {arg}} has negative elements, check to ensure that values",
+        "have not been log transformed",
```

Comment:
```suggestion
      c("{.arg {arg}} has negative elements.",
```
My knee-jerk reaction is that the "log transformed" bit is too specific given the generic function name and internal logic. 

I understand that in practice this assertion will often be applied to quantities such as concentrations per mL, which are often recorded in log units but which we expect in natural units. Even then, though, that's not the only way a concentration vector could accidentally have negative values.

If you really want detailed messages for concentrations, I'd make it possible to configure the error message and add concentration-specific details to the error when it is being used to check concentrations, e.g. something like

```R
assert_elements_non_neg <- function(x, arg = "x", call = rlang::caller_env(), additional_error_msg="")
```
and then something like
```R
cli::cli_abort(
      c("{.arg {arg}} has negative elements. ", additional_error_msg,
      "!" = "All elements must be 0 or greater",
      "i" = "Elements {.val {which(!is_non_neg)}} are negative"),
      class = "wwinference_input_data_error",
      call = call)
```

and then something like:

```R
assert_elements_non_neg(concentration_vector, additional_error_message="Note that the model expects natural scale concentration values, which must be non-negative.")

Reply:
I like this a lot and think it makes sense, bc I can write these into the `validate` functions. Will go ahead and make this change.